GVFS/GVFS.FunctionalTests/Tests/DiskLayoutVersionTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿using GVFS.FunctionalTests.Tests.EnlistmentPerTestCase;
+using GVFS.FunctionalTests.Tools;
+using GVFS.Tests.Should;
+using NUnit.Framework;
+using System.Runtime.InteropServices;
+
+namespace GVFS.FunctionalTests.Windows.Tests
+{
+    [TestFixture]
+    [Category(Categories.FullSuiteOnly)]
+    [Category(Categories.WindowsOnly)]
+    public class DiskLayoutVersionTests : TestsWithEnlistmentPerTestCase
+    {
+        private const int WindowsCurrentDiskLayoutMajorVersion = 17;
+        private const int MacCurrentDiskLayoutMajorVersion = 18;
+        private const int CurrentDiskLayoutMinorVersion = 0;
+        private int currentDiskMajorVersion;
+
+        [SetUp]
+        public override void CreateEnlistment()
+        {
+            base.CreateEnlistment();
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                this.currentDiskMajorVersion = MacCurrentDiskLayoutMajorVersion;
+            }
+            else
+            {
+                this.currentDiskMajorVersion = WindowsCurrentDiskLayoutMajorVersion;
+            }
+        }
+
+        [TestCase]
+        public void MountSucceedsIfMinorVersionHasAdvancedButNotMajorVersion()
+        {
+            // Advance the minor version, mount should still work
+            this.Enlistment.UnmountGVFS();
+            GVFSHelpers.SaveDiskLayoutVersion(
+                this.Enlistment.DotGVFSRoot,
+                this.currentDiskMajorVersion.ToString(),
+                (CurrentDiskLayoutMinorVersion + 1).ToString());
+            this.Enlistment.TryMountGVFS().ShouldBeTrue("Mount should succeed because only the minor version advanced");
+
+            // Advance the major version, mount should fail

```

Comment:
Could you also added a test that mount will fail if the version that's on disk is lower than the minimum supported version (7 on Windows and 18 on Mac)?